ImportError: urllib3 v2.0 only supports OpenSSL 1.1.1+

# Bug Report
Importing Synapseclient on Linux system causes an import Error if LibreSSL is used instead of OpenSSL.

## Operating system
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

## Client version
Python 3.9.0
Synapse 2.7.0

## Description of the problem
Importing Synapseclient on Linux system causes an import Error if LibreSSL is used instead of OpenSSL.
Can be hotfixed by installing an older version of `urllib3`, but that is not the ideal solution going forward, I'd assume.

### Expected behavior

No error ;)

### Actual behavior
```
.local/lib/python3.9/site-packages/synapseclient/__init__.py:275: in <module>
    from .activity import Activity
.local/lib/python3.9/site-packages/synapseclient/activity.py:74: in <module>
    from synapseclient.core.exceptions import SynapseError, SynapseMalformedEntityError
.local/lib/python3.9/site-packages/synapseclient/core/exceptions.py:15: in <module>
    import requests
.local/lib/python3.9/site-packages/requests/__init__.py:43: in <module>
    import urllib3
.local/lib/python3.9/site-packages/urllib3/__init__.py:38: in <module>
    raise ImportError(
E   ImportError: urllib3 v2.0 only supports OpenSSL 1.1.1+, currently the 'ssl' module is compiled with LibreSSL 3.2.1. See: https://github.com/urllib3/urllib3/issues/2168
```

--- a/setup.py
+++ b/setup.py
@@ -42,7 +42,7 @@ test_deps = [
 ]
 
 install_requires = [
-    "requests>=2.22.0,<3.0",
+    "requests>=2.24.0,<3.0",
     "urllib3<2; python_version<'3.10'",
     "urllib3>=2; python_version>='3.10'",
     "keyring>=15,<23.5",

--- a/setup.py
+++ b/setup.py
@@ -42,7 +42,8 @@ test_deps = [
 ]
 
 install_requires = [
-    "requests>=2.24.0,<3.0",
+    "requests>=2.22.0,<2.30.0; python_version<'3.10",
+    "requests>=2.30.0,<3.0; python_version>='3.10",
     "urllib3<2; python_version<'3.10'",
     "urllib3>=2; python_version>='3.10'",
     "keyring>=15,<23.5",

--- a/setup.py
+++ b/setup.py
@@ -43,6 +43,7 @@ test_deps = [
 
 install_requires = [
     "requests>=2.22.0,<3.0",
+    "urllib3<2",
     "keyring>=15,<23.5",
     "deprecated>=1.2.4,<2.0",
     "importlib-metadata<5.0",

--- a/setup.py
+++ b/setup.py
@@ -43,7 +43,8 @@ test_deps = [
 
 install_requires = [
     "requests>=2.22.0,<3.0",
-    "urllib3<2",
+    "urllib3<2; python_version<'3.10'",
+    "urllib3>=2; python_version>='3.10'",
     "keyring>=15,<23.5",
     "deprecated>=1.2.4,<2.0",
     "importlib-metadata<5.0",

--- a/setup.py
+++ b/setup.py
@@ -42,10 +42,10 @@ test_deps = [
 ]
 
 install_requires = [
-    "requests>=2.22.0,<2.30.0; python_version<'3.10'",
-    "requests>=2.30.0,<3.0; python_version>='3.10'",
-    "urllib3<2; python_version<'3.10'",
-    "urllib3>=2; python_version>='3.10'",
+    # "requests>=2.22.0,<2.30.0; python_version<'3.10'",
+    "requests>=2.22.0,<3.0",
+    "urllib3<2",
+    # "urllib3>=2; python_version>='3.10'",
     "keyring>=15,<23.5",
     "deprecated>=1.2.4,<2.0",
     "importlib-metadata<5.0",

--- a/setup.py
+++ b/setup.py
@@ -42,8 +42,8 @@ test_deps = [
 ]
 
 install_requires = [
-    "requests>=2.22.0,<2.30.0; python_version<'3.10",
-    "requests>=2.30.0,<3.0; python_version>='3.10",
+    "requests>=2.22.0,<2.30.0; python_version<'3.10'",
+    "requests>=2.30.0,<3.0; python_version>='3.10'",
     "urllib3<2; python_version<'3.10'",
     "urllib3>=2; python_version>='3.10'",
     "keyring>=15,<23.5",

--- a/synapseclient/table.py
+++ b/synapseclient/table.py
@@ -1879,6 +1879,7 @@ class TableQueryResult(TableAbstractBaseClass):
             offset += len(self.rowset['rows'])
 
             if not rowIdAndVersionInIndex:
+                # TODO: Look into why this isn't being assigned
                 series['ROW_ID'].append(pd.Series(name='ROW_ID', data=[row['id'] for row in self.rowset['rows']]))
                 series['ROW_VERSION'].append(pd.Series(name='ROW_VERSION',
                                                        data=[row['version'] for row in self.rowset['rows']]))
@@ -1888,10 +1889,13 @@ class TableQueryResult(TableAbstractBaseClass):
 
             for i, header in enumerate(self.rowset["headers"]):
                 column_name = header.name
-                series[column_name] = series[column_name].append(
-                    pd.Series(name=column_name,
-                              data=[row['values'][i] for row in self.rowset['rows']],
-                              index=rownames),
+                series[column_name] = pd.concat(
+                    [
+                        series[column_name],
+                        pd.Series(name=column_name,
+                                data=[row['values'][i] for row in self.rowset['rows']],
+                                index=rownames)
+                    ],
                     # can't verify integrity when indices are just numbers instead of 'rowid_rowversion'
                     verify_integrity=rowIdAndVersionInIndex)
 

--- a/synapseclient/table.py
+++ b/synapseclient/table.py
@@ -1893,8 +1893,8 @@ class TableQueryResult(TableAbstractBaseClass):
                     [
                         series[column_name],
                         pd.Series(name=column_name,
-                                data=[row['values'][i] for row in self.rowset['rows']],
-                                index=rownames)
+                                  data=[row['values'][i] for row in self.rowset['rows']],
+                                  index=rownames)
                     ],
                     # can't verify integrity when indices are just numbers instead of 'rowid_rowversion'
                     verify_integrity=rowIdAndVersionInIndex)

--- a/synapseutils/sync.py
+++ b/synapseutils/sync.py
@@ -166,7 +166,7 @@ class _FolderSync:
                     )
 
                 # in practice only the root folder sync will be waited on/need notifying
-                self._finished.notifyAll()
+                self._finished.notify_all()
 
     def _manifest_filename(self):
         return os.path.expanduser(
@@ -195,7 +195,7 @@ class _FolderSync:
                 self._parent.set_exception(exception)
 
             # an error also results in the folder being finished
-            self._finished.notifyAll()
+            self._finished.notify_all()
 
     def wait_until_finished(self):
         with self._finished:
@@ -596,7 +596,7 @@ class _SyncUploader:
                         # this item was defined as provenance for another item, now that
                         # it's finished we may be able to upload that depending item, so
                         # wake up he central thread
-                        dependency_condition.notifyAll()
+                        dependency_condition.notify_all()
 
                     except KeyError:
                         # this item is not used in provenance of another item, that's fine
@@ -605,7 +605,7 @@ class _SyncUploader:
         except Exception:
             with dependency_condition:
                 abort_event.set()
-                dependency_condition.notifyAll()
+                dependency_condition.notify_all()
             raise
 
         finally:

--- a/tests/unit/synapseutils/unit_test_synapseutils_sync.py
+++ b/tests/unit/synapseutils/unit_test_synapseutils_sync.py
@@ -770,7 +770,7 @@ class TestSyncUploader:
         assert len(pending_provenance._pending) == 0
 
         # should have notified the condition to let anything waiting on this provenance to continue
-        mock_condition.notifyAll.assert_called_once_with()
+        mock_condition.notify_all.assert_called_once_with()
 
         assert not abort_event.is_set()
 
@@ -815,7 +815,7 @@ class TestSyncUploader:
         # abort event should have been raised and we shoudl have released threading locks
         assert abort_event.is_set()
         mock_release.assert_called_once_with()
-        mock_condition.notifyAll.assert_called_once_with()
+        mock_condition.notify_all.assert_called_once_with()
 
     def test_abort(self):
         """Verify abort behavior.
@@ -927,7 +927,7 @@ class TestSyncUploader:
                 convert_provenance_calls -= 1
 
                 if convert_provenance_calls == 0:
-                    convert_provenance_condition.notifyAll()
+                    convert_provenance_condition.notify_all()
 
             return convert_provenance_original(provenance, finished_items)
 

